### ISSUE TITLE
Host Unreachable

### ISSUE THREAD

**Issue by Debojit** [motdotla/ansible-pi#7]
Hi,

I'm running this on macOS 10.12.5, python 2.7.13, ansible 2.3.0.0. I get this error when I try to run the playbook:

$ ansible-playbook playbook.yml -i hosts --ask-pass --become -c paramiko
SSH password: 

PLAY [Ansible Playbook for configuring brand new Raspberry Pi] *********************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************************************************
fatal: [192.168.1.xxx]: UNREACHABLE! => {"changed": false, "msg": "[Errno None] Unable to connect to port 22 on 192.168.1.xxx", "unreachable": true}
	to retry, use: --limit @/Users/[username]/Downloads/ansible-pi/playbook.retry

PLAY RECAP *************************************************************************************************************************************************************************************************
192.168.1.xxx               : ok=0    changed=0    unreachable=1    failed=0

**Comment by Nishanth-R** [comment]
Hi , 
Facing the same issue : For traceability , I have added the Raspi's ip address in Hosts file, updated the wpa_supplicant.conf file and ran the command. Am I missing something?

**Comment by Nishanth-R** [comment]
Figured out what the Issue was , Raspbian switches off SSH by default now and hence this project wont work until you connect to peripherals and enable ssh.

**Comment by davidnquach** [comment]
You can just add an ssh file to the boot path 

> If you want to enable SSH, all you need to do is to put a file called ssh in the /boot/ directory. The contents of the file don’t matter: it can contain any text you like, or even nothing at all. When the Pi boots, it looks for this file; if it finds it, it enables SSH and then deletes the file.

ref: https://www.raspberrypi.org/blog/a-security-update-for-raspbian-pixel/

**Comment by Marto32** [comment]
the guide that @davidnquach posted worked for me regarding the connection failure, but now I'm getting this error:

```
{"changed": false, "msg": "host key mismatch for 192.168.1.xxx", "unreachable": true}
```

Any idea how to rectify this?

I'm not sure what to enter for the `SSH Password` and `SSH Key for your new device` prompts.

**Comment by davidnquach** [comment]
@Marto32 

If this is your first time setting up; the default user is pi, and the password is raspberry.

ref: https://www.raspberrypi.org/documentation/linux/usage/users.md

**Comment by Marto32** [comment]
Thanks @davidnquach. I think the problem is the ssh key prompt. What do you enter for `please enter the SSH Key for your new device::`? 

I've tried blank and the one from my `known_hosts` file but I keep getting this error now:

```
fatal: [192.168.1.xxx]: UNREACHABLE! => {"changed": false, "msg": "Authentication failed.", "unreachable": true}
```

**Comment by davidnquach** [comment]
I don't believe I have run into that but have you generated an ssh key for your local machine? Also how are you running the ssh command? `ssh pi@192.168.0.xxx` should just prompt you for the password and another prompt for you to agree that it is a safe machine to log into.

**Comment by Marto32** [comment]
Got it working - thank you!
